common/fs/src/rule.rs

```diff
@@ -1,20 +1,47 @@
 use std::fmt::Debug;
 use std::path::Path;
-use std::str::FromStr;
 
 use globber::{Error as PatternError, Pattern};
 use pcre2::{bytes::Regex, Error as RegexError};
 use std::os::unix::ffi::OsStrExt;
 
-/// A list of rules
-pub type RuleList = Vec<Box<dyn Rule + Send>>;
-
 /// A trait for implementing a rule, see GlobRule/RegexRule for an example
-pub trait Rule: Debug {
+pub trait Rule {
     /// Takes a value and returns true or false based on if it matches
     fn matches(&self, value: &Path) -> bool;
 }
 
+#[derive(Debug, Clone)]
+pub enum RuleDef {
```

Comment:
To make it easier to clone the values, I turned them into an enum from a trait object.

Reply:
The branch to pick an implementation is also probably cheaper than chasing the pointer on every element when iterating the rules. 👍